app/assets/scripts/views/location/measurements-card.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { PropTypes as T } from 'prop-types';
+import _ from 'lodash';
+import moment from 'moment';
+import styled from 'styled-components';
+import LoadingMessage from '../../components/loading-message';
+import Card, { HighlightText, CardSubtitle } from '../../components/card';
+
+const StyledLoading = styled(LoadingMessage)`
+  grid-column: 4 / 11;
+`;
+const ErrorMessage = styled.div`
+  grid-column: 4 / 11;
+`;
+
+const MeasurementsCont = styled.dl`
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(5rem, 1fr));
+  grid-gap: 0.25rem;
+  height: 60%;
+  overflow-y: scroll;
+`;
+const Measurement = styled.div`
+  width: min-content;
+`;
+
+export default function Measurements({
+  latestMeasurements,
+  measurements,
+  loc,
+  parameters,
+}) {
+  const {
+    fetched: lastMFetched,
+    fetching: lastMFetching,
+    error: lastMError,
+    data: { results: lastMeasurements },
+  } = latestMeasurements;
+  const {
+    fetched: mFetched,
+    fetching: mFetching,
+    error: mError,
+  } = measurements;
+
+  const error = lastMError || mError;
+  const fetched = lastMFetched || mFetched;
+  const fetching = lastMFetching || mFetching;
+
+  if (!fetched && !fetching) {
+    return null;
+  }
+
+  if (!fetched && !fetching) {
+    return null;
+  } else if (fetching) {
+    return <StyledLoading />;
+  } else if (error) {
+    return (
+      <ErrorMessage className="fold__introduction prose prose--responsive">
+        <p>{"We couldn't get stats. Please try again later."}</p>
+        <p>
+          {"If you think there's a problem, please "}
+          <a href="mailto:info@openaq.org" title="Contact openaq">
+            contact us.
+          </a>
+        </p>
+      </ErrorMessage>
+    );
+  }
+
+  let locData = loc.data;
+
+  // Get latest measurements for this location in particular.
+  let locLastMeasurement = _.find(lastMeasurements, {
+    location: locData.location,
+  });
+  console.log(locLastMeasurement.measurements);
+
+  return (
+    <Card
+      title="Latest Measurements"
+      gridColumn={'4 / 11'}
+      // gridRow={'1 / span 2'}
+      renderBody={() => {
+        return (
+          <div className="card__body">
+            {locLastMeasurement ? (
+              <MeasurementsCont>
+                {locLastMeasurement.measurements.reduce((acc, o) => {
```

Comment:
since these should cards render before the content returns (display loading component), i advocate for leaving the reduction on the card side. what do you think @AliceR